### DISCUSSION TITLE
Add item with results

### DISCUSSION THREAD

**Discussion by Hugo-Matias** [Blazored/Typeahead#304] [Q&A]
I'm implementing an AddItemOnEmptyResult when my search comes empty but I would like to be able to add a new item even if some data is retrieved but not an exact match. Is there a workaround to display the NotFoundTemplate even when some data is retrieved from the SearchMethod?

My search function uses .Contains() to retrieve similar objects but in essence I would like to add an item if .Equals() returns empty while keep using contains for the search method.

For instance:

**Input:** foo
**Results:** 
foobar

I would like to set it like so:

**Input:** foo
**Results:** 
foobar
Add new entry: foo _<- NotFoundTemplate_

**Comment by chrissainty** [comment]
Have you tried using the FooterTemplate?

**Reply by Hugo-Matias** [reply]
Thanks for taking the time to reply Chris, really appreciate it.

That does look like what I need in this case (somehow missed it while reading the docs), but trying to set it up, it appears to be missing the context parameter to use the input as a new value. Should we use a method for a dynamic footer to behave like AddItemOnEmptyResultMethod?
